plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/deploy/_new/NewApplicationPresenter.java

```diff
@@ -103,103 +252,408 @@ public void onCancelClicked() {
+                List<ContainerPort> ports = parsePorts(exposedPorts);
+
+                String namespace = project.getMetadata().getName();
+
+                Promises.all(osService.createImageStream(generateImageStream(namespace, labels)),
+                             osService.createBuildConfig(generateBuildConfig(namespace, labels)),
+                             osService.createDeploymentConfig(generateDeploymentConfig(namespace, labels)),
+                             osService.createRoute(generateRoute(namespace, labels)),
+                             osService.createService(generateService(namespace, getFirstPort(ports), labels)))
+                        .then(new Operation<JsArrayMixed>() {
+                            @Override
+                            public void apply(JsArrayMixed arg) throws OperationException {
+                                view.hide();
+                                notificationManager
+                                        .showInfo(locale.deployProjectSuccess(appContext.getCurrentProject().getRootProject().getName()));
+                                setupMixin(project);
+                            }
+                        }).catchError(new Operation<PromiseError>() {
+                    @Override
+                    public void apply(PromiseError arg) throws OperationException {
+                        //TODO remove notification, show error only on popup:
+                        final ServiceError serviceError = dtoFactory.createDtoFromJson(arg.getMessage(), ServiceError.class);
+                        notificationManager.showError(serviceError.getMessage());
+                        view.showError(serviceError.getMessage());
+                    }
+                });
+            }
+        });
     }
 
-    @Override
-    public void onProjectNameChanged() {
+    private void setupMixin(Project project) {
+        List<String> mixins = appContext.getCurrentProject().getRootProject().getMixins();
+        if (!mixins.contains(OPENSHIFT_PROJECT_TYPE_ID)) {
+            mixins.add(OPENSHIFT_PROJECT_TYPE_ID);
+        }
 
+        Map<String, List<String>> attributes = appContext.getCurrentProject().getRootProject().getAttributes();
+        attributes.put(OPENSHIFT_APPLICATION_VARIABLE_NAME, newArrayList(osAppName));
+        attributes.put(OPENSHIFT_NAMESPACE_VARIABLE_NAME, newArrayList(project.getMetadata().getName()));
+        final ProjectConfigDto updateProject = newDto(ProjectConfigDto.class)
+                .withMixins(mixins)
+                .withType(appContext.getCurrentProject().getRootProject().getType())
+                .withAttributes(attributes);
+
+        newPromise(new RequestCall<ProjectDescriptor>() {
+            @Override
+            public void makeCall(AsyncCallback<ProjectDescriptor> callback) {
+                projectService.updateProject(appContext.getCurrentProject().getRootProject().getPath(),
```

Comment:
It would be better to use deprecated update project https://github.com/codenvy/che-plugins/blob/openshift/plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/UnlinkProjectAction.java#L88 for now